addon/event_dispatcher.js

```diff
@@ -128,51 +147,27 @@ export default EventDispatcher.extend({
       events.dblclick = null;
     }
 
-    // assign custom events into Ember's Eventing
-    assign(events, addedEvents || {});
-
     // delete unwanted events
-    for (event in events) {
-      if (events.hasOwnProperty(event)) {
-        if (!events[event]) {
-          delete events[event];
-        }
+    for (let event in events) {
+      if (events.hasOwnProperty(event) && !events[event]) {
+        delete events[event];
       }
     }
 
-    // assign our events into Ember's Eventing
-    assign(events, this.get('_gestures'));
-
-    if (!isNone(rootElement)) {
-      set(this, 'rootElement', rootElement);
-    }
-    this._fastFocus();
-
-
-    rootElement = jQuery(get(this, 'rootElement'));
-
-    assert(`You cannot use the same root element (${rootElement.selector || rootElement[0].tagName}) multiple times in an Ember.Application`, !rootElement.is(ROOT_ELEMENT_SELECTOR));
-    assert('You cannot make a new Ember.Application using a root element that is a descendent of an existing Ember.Application', !rootElement.closest(ROOT_ELEMENT_SELECTOR).length);
-    assert('You cannot make a new Ember.Application using a root element that is an ancestor of an existing Ember.Application', !rootElement.find(ROOT_ELEMENT_SELECTOR).length);
+    // override default events
+    this.set('events', events);
 
-    rootElement.addClass(ROOT_ELEMENT_CLASS);
+    // add our events to addition events
+    let additionalEvents = assign({}, addedEvents);
+    additionalEvents = assign(additionalEvents, this.get('_gestures'));
 
-    assert(`Unable to add '${ROOT_ELEMENT_CLASS}' class to rootElement. Make sure you set rootElement to the body or an element in the body.`, rootElement.is(ROOT_ELEMENT_SELECTOR));
-
-    // needed for Ember 2.12+ support (setupHandler takes 4 args)
-    let viewRegistry = this._getViewRegistry && this._getViewRegistry();
+    this._fastFocus();
 
-    for (event in events) {
-      if (events.hasOwnProperty(event)) {
-        this.setupHandler(rootElement, event, events[event], viewRegistry);
-      }
-    }
+    return this._super(additionalEvents, rootElement);
```

Comment:
Note: I removed quite a bit of code duplicated from the original `EventDispatcher`, by calling its super method here, instead of overriding and duplicating everything.

Reply:
this is great work -- just so i fully understand - 

previously we called `assign(events, this.get('_gestures'));` then `this._fastFocus();`, and then registered each event, including those added via `this.get('_gestures')` --- now we're calling `this.set('events', events);` then `this._fastFocus();` and then calling `return this._super(additionalEvents, rootElement);` to allow the ember event dispatcher to do the `for (event in events) {... this.setupHandler ...` for us, correct? 

If I'm missing something, please let me know.

One more q, where'd this guy go?

```
$root.on('click.ember-gestures touchend.ember-gestures', function (e) {
```

other than that, at least with just an eyeball review on github here, it looks like you nailed it... is there any way u could bring a piece of code from ur app into a test? (there isn't really much testing now.. so u can provide another way to prove its correctness) -- basically anything that will verify the new behavior matches the previous behavior, without careful reading through the diff here, would be nice, in some form, even logging the events registered through ember vs those on the rootElement itself,  via `this._fastFocus()`, aka, those not using the previous verbose `this.setupHandler` loop -- would ease any concerns I have over my own ability to read and understand this code, old and new :) 

also @runspired should approve this as he wrote it originally... i think i made any concerns obvious, the rest checks out except the ordering where mentioned, and the missing `.on('click...` from jquery (maybe i just missed it).

Reply:
@eriktrom the changes here lived in fork I created a while ago, so I have to refresh my memories a bit... 😬

> to allow the ember event dispatcher to do the for (event in events) {... this.setupHandler ... for us, correct?

Yes, exactly. `setup()` supports a list of additional events, so I just use that to pass the gestures specific events to the parent to let it attach the event listeners for them.

> One more q, where'd this guy go?

This adds `click` and `touchend` listeners via jQuery, and the same happens here: 
https://github.com/html-next/ember-gestures/pull/114/files#diff-89ef10d070270d2503408f9f16d8b558R91
for the `fastFocusEvents` here: https://github.com/html-next/ember-gestures/pull/114/files#diff-89ef10d070270d2503408f9f16d8b558R27

> is there any way u could bring a piece of code from ur app into a test

Hm, not really. These are app specific (lots of acceptance and component integration tests), which would certainly have failed indirectly if the EventDispatcher was not working correctly. I also verified that some gestures we used like swipes were still working.

For adding some real tests here, I am afraid to not have enough time for this atm, to do it properly. :(